### ISSUE TITLE
The id of `customIpPrefix` with customIpPrefix name in uppercase returned by 'Public IP Prefixes - Create Or Update' API

### ISSUE THREAD

**Issue by sinbai** [Azure/azure-rest-api-specs#17059]
The id of `customIpPrefix` with customIpPrefix name in uppercase returned by '[Public IP Prefixes - Create Or Update' API](https://github.com/Azure/azure-rest-api-specs/blob/c09b26d4aaa40fb37380aae01b7019b75cff3e34/specification/network/resource-manager/Microsoft.Network/stable/2021-03-01/publicIpPrefix.json#L152) :

Repro steps:

Create a publicIpPrefix resource (specify `customIPPrefix` property)
The id of "customIpPrefix" in the following request body:
```
   "customIPPrefix": {      
        "id": /subscriptions/XXX/resourceGroups/test/providers/Microsoft.Network/customIpPrefixes/testpip01"
      }
```

The id of "customIPPrefix" returned in the following response body:

```
 "customIpPrefix": {
      "id": "/subscriptions/XXX/resourceGroups/test/providers/Microsoft.Network/customIPPrefixes/testpip01"
    }
```

Expected:
The id of `customIpPrefix` with custom Ip Prefix name should be `customIpPrefixes`.

Actual:
The id of `customIpPrefix` with custom Ip Prefix name is `customIPPrefixes`.

**Comment by tombuildsstuff** [comment]
it's worth calling out that the JSON keys aren't the same here too?

**Comment by sinbai** [comment]
> it's worth calling out that the JSON keys aren't the same here too?

Yes, the JSON keys are not the same too. The `customIPPrefix` in request body but the `customIpPrefix` returned by the ['Public IP Prefixes - Create Or Update' API](https://github.com/Azure/azure-rest-api-specs/blob/c09b26d4aaa40fb37380aae01b7019b75cff3e34/specification/network/resource-manager/Microsoft.Network/stable/2021-03-01/publicIpPrefix.json#L152).
